### PR TITLE
AS: added jsdoc comments

### PR DESCRIPTION
**Pull request by Abdul-NHS**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: terraform/src/gtmsMeshMailboxLambda/gtmsMeshMailboxLambda.js [hunk 1]
```diff
@@ -90,17 +90,32 @@ export const handler = async (event, context) => {
...
- * The outbound-gtms-invited-participant-batch bucket is not included here as it is
- * an outbound bucket, where GPS will be sending info to GTMS
+/**
+ * Processing incoming messages from GTMS (MESH mailbox) based on workflow ids
+ *
+ * @function processMessage
+ * @async
+ * @param {Object} message - The payload received
+ * @param {string} environment - Which environment the resources are allocated to
+ * @param {Object} S3client - An instance of S3 client
```
Comment (chenglawson1) [inline terraform/src/gtmsMeshMailboxLambda/gtmsMeshMailboxLambda.js:115]: Could change Object to S3Client.

File: terraform/src/nrdsMeshMailboxLambda/helper.js [hunk 2]
```diff
@@ -34,14 +60,16 @@ export const getSecret = async (secretName, client) => {
...
-}
+};
 
-/*
+/**
  * Establish connection with MESH
- * @params handshake to be replaced with handShake fn
+ *
+ * @param {Object} CONFIG - Configuration obj includes mailbox id, certs etc
+ * @param {string} handshake - Replace handshake with handShake fn
```
Comment (chenglawson1) [inline terraform/src/nrdsMeshMailboxLambda/helper.js:69]: Is handshake a function so the param type should be function?

File: terraform/src/nrdsMeshMailboxLambda/helper.js [hunk 3]
```diff
@@ -55,11 +83,14 @@ export const getHealthStatusCode = async (CONFIG, handshake) => {
...
     console.error(`Error occurred: ${error}`);
   }
-}
+};
 
-/*
+/**
  * Return an array of message IDs from MESH
- * @params msgCount to be replaced with getMessageCount fn
+ *
```
Comment (chenglawson1) [inline terraform/src/nrdsMeshMailboxLambda/helper.js:90]: Please add @async.

File: terraform/src/nrdsMeshMailboxLambda/helper.js [hunk 4]
```diff
@@ -55,11 +83,14 @@ export const getHealthStatusCode = async (CONFIG, handshake) => {
...
+ * @param {Object} CONFIG - Configuration obj includes mailbox id, certs etc
+ * @param {string} msgCount - Replace msgCount with getMessageCount fn
```
Comment (chenglawson1) [inline terraform/src/nrdsMeshMailboxLambda/helper.js:92]: msgCount is a function?

File: terraform/src/nrdsMeshMailboxLambda/helper.js [hunk 5]
```diff
@@ -80,12 +111,15 @@ export const getMessageArray = async (CONFIG, msgCount) => {
...
-/*
- * Marks messaged as read based on the message ID passed in,
- * subsequently removing message from MESH mailbox
- * @params marked to be replaced with markAsRead fn
+/**
+ * Marks messaged as read based on the message ID passed in, subsequently removing message from MESH mailbox
+ *
+ * @param {*} CONFIG - Configuration obj includes mailbox id, certs etc
+ * @param {*} marked - Replace marked with markAsRead fn
+ * @param {*} msgID - Message you want to querying
```
Comment (chenglawson1) [inline terraform/src/nrdsMeshMailboxLambda/helper.js:121]: Is CONFIG an Object?
Is marked a function?
Is msgID a string?

File: terraform/src/nrdsMeshMailboxLambda/helper.js (hunk 2 above)
Comment (Abdul-NHS) [reply terraform/src/nrdsMeshMailboxLambda/helper.js]: yes, resolved

File: terraform/src/nrdsMeshMailboxLambda/helper.js (hunk 4 above)
Comment (Abdul-NHS) [reply terraform/src/nrdsMeshMailboxLambda/helper.js]: resolved

File: terraform/src/nrdsMeshMailboxLambda/helper.js (hunk 5 above)
Comment (Abdul-NHS) [reply terraform/src/nrdsMeshMailboxLambda/helper.js]: Was wip, just adding the other lambda in now too. Resolved

**Review by chenglawson1 — APPROVED** [review]

**Review by avi312NHS — APPROVED** [review]

**Pull request merged**